configure.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Top level configure script conditionally calling more specific configuration
+scripts"""
+
+import pathlib
+import shlex
+import subprocess
+
+from shutil import which
+
+
+def run_shell(cmd_str, stdout=None):
+  cmd = shlex.split(cmd_str)
+  cmd_rslt = subprocess.run(cmd, check=True, text=True, stdout=stdout)
+
+  if cmd_rslt.returncode != 0:
+    raise RuntimeError(f"[ERROR] Command `{cmd_str}` returned with code: "
+                       f"{cmd_rslt.returncode}")
+
+  if stdout is not None:
+    return cmd_rslt.stdout
+
+
+def _get_input(question, default_answer):
+  try:
+    answer = input(f"\n- {question} ")
+  except EOFError:
+    answer = default_answer
+
+  return (answer or default_answer).strip().lower()
+
+
+def get_input(question, default_answer='', accepted_answers=None):
+  if accepted_answers is None:
```

Comment:
Actually seems like we will likely have cases where there isn't a pre-defined set of accepted answers and None would be the logical sentinel for that case